pkg/daemon/cleanup/cleanup.go

```diff
@@ -57,13 +57,37 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 		klog.Exitf("could not create service manager: %s", err.Error())
 	}
 	defer svcMgr.Disconnect()
-	services, err := getServicesToRemove(ctx, directClient, node, configMapNamespace)
+	var relevantCM *core.ConfigMap
+	var services []servicescm.Service
+	// get data from the latest services ConfigMap
+	latestCM, err := servicescm.GetLatest(directClient, ctx, configMapNamespace)
 	if err != nil {
-		return err
+		return fmt.Errorf("cannot get latest services ConfigMap from namespace %s: %w", configMapNamespace, err)
+	}
+	versionCM, err := servicescm.GetVersionCM(ctx, directClient, node, configMapNamespace)
+	if err != nil {
+		klog.Infof("error getting services ConfigMap associated with version annotation, "+
+			"falling back to use latest services ConfigMap: %s", err)
+		relevantCM = latestCM
+		latestCMData, err := servicescm.Parse(latestCM.Data)
+		if err != nil {
+			return err
+		}
+		services = latestCMData.Services
+	} else {
+		relevantCM = versionCM
+		services, err = getServicesToRemove(latestCM, versionCM)
+		if err != nil {
+			return err
+		}
```

Comment:
Its not clear why this logic has to be done, which indicates it should be broken out into a separate documented function.